R/getNeatOrder.R

```diff
@@ -0,0 +1,179 @@
+        ...){
+              
+            # Check args
+            .check_args(x, subset, dimensions, centering_method, decreasing)
+            
+            # Create subset if required
+            if( !is.null(subset) ){
+                x <- x[subset, , drop = FALSE]
+            }
+            
+            # Take the correct dimensions
+            x <- x[, dimensions, drop = FALSE]
+
+            # Get the theta values and order them
+            theta_values <- .radial_theta(x, centering_method)
+            ordering <- .get_sorted_rownames(theta_values, decreasing)
+            
+            return(ordering)
+        }
+    )
+
+
+# Checks the method arguments.
+.check_args <- function(x, subset, dimensions, centering_method, decreasing) {
+    # Check data is a matrix
+    if (!is.matrix(x)) {
+        stop("Input data must be a matrix.", call. = FALSE)
+    }
+    
+    # Check there is sufficient data
+    if (nrow(x) == 0 || ncol(x) == 0) {
+        stop("No data to plot. Matrix must have at least one row and one column.", call. = FALSE)
+    }
+    
+    # Check subset validity
+    if (!is.null(subset)) {
+        if (is.numeric(subset) && any(subset > nrow(x))) {
+            stop("Subset refers to rows that do not exist in the data.", call. = FALSE)
+        } else if (is.character(subset) && any(!subset %in% rownames(x))) {
+            stop("Subset refers to row names that do not exist in the data.", call. = FALSE)
+        } else if (!is.numeric(subset) && !is.character(subset)) {
+            stop("Subset must be a vector of row indices or names.", call. = FALSE)
+        }
+    }
+    
+    # Check dimensions are valid
+    if (is.numeric(dimensions)) {
+        if (any(dimensions > ncol(x) | dimensions < 1)) {
+            stop("dimensions refer to columns that do not exist in the data.", call. = FALSE)
+        }
+    } else if (is.character(dimensions)) {
+        if (any(!dimensions %in% colnames(x))) {
+            stop("dimensions refer to column names that do not exist in the data.", call. = FALSE)
+        }
+    } else {
+        stop("dimensions must be a vector of column indices or names.", call. = FALSE)
+    }
+    
+    # Check dimension vector is of length 2
+    if (length(dimensions) != 2) {
+        stop("Exactly two dimensions must be specified.", call. = FALSE)
+    }
+    
+    # Check centering_method
+    centering_method <- match.arg(centering_method, c("mean", "median", "none"))
+    
+    # Check decreasing
+    if (!is.logical(decreasing) || length(decreasing) != 1) {
+        stop("decreasing must be a single boolean value.", call. = FALSE)
+    }
+    
+    # Check for unique row names
+    if (any(duplicated(rownames(x)))) {
+        stop("Row names of the matrix must be unique.", call. = FALSE)
+    }
+    
+    # Check for unique column names
+    if (any(duplicated(colnames(x)))) {
+        stop("Column names of the matrix must be unique.", call. = FALSE)
+    }
```

Comment:
These input check could be added directly to "main" code since there are not too many parameters to check